### PR TITLE
Nodejs integration test improvements

### PR DESCRIPTION
**Pull request by ChrisDufourMB**
## Fixes

- Sets up the node example project in a way developers can set up simply by running `npm i && npm test`
- Small improvements to nodeJS integration tests

## Description

Adds some small improvements to the nodeJS integration tests
Improvements to the setup of the nodeJS example project

## Proposed changes in this PR

- Updates `package.json` and adds a `package-lock.json` file to the nodeJS example project
- Small improvements to nodeJS build/test script

## Things to look at

- [x] Test coverage
- [x] Code Style
- [x] Documentation (READMEs etc)

### CODE REVIEW

File: examples/node/package.json [hunk 1]
```diff
@@ -7,5 +7,10 @@
     "test": "tsc -p tsconfig.json && node node.js"
   },
   "author": "@gamebridgeai",
-  "license": "ISC"
+  "license": "ISC",
+  "dependencies": {
+    "@gamebridgeai/ts_serialize": "0.2.4",
```
Comment (hardy925) [inline examples/node/package.json]:
```suggestion
    "@gamebridgeai/ts_serialize": "*",
```

I think this always grabs the latest so we don't need to manually update for releases, maybe we add it for all our packages here.

Thoughts.

**Review by hardy925 — COMMENTED** [review]
nice!

File: examples/node/package.json (hunk 1 above)
Comment (ChrisDufourMB) [reply examples/node/package.json]: Sounds good. We currently don't have a way to automatically update the lockfile, but TBH I don't think it's a big deal b/c we don't need to update the dependency for `ts_serialize`.

The lockfile it's self is ignored unless explicitly added so we shouldn't have any issue with people accidentally committing an updated lockfile. The tests links the build release, so even if the lockfile is pointing to an aincent version of ts_serialize we shouldn't have any problems.

**Review by hardy925 — APPROVED** [review]
awesome update

**Pull request merged**